src/groupCurrencyToken.ts

```diff
@@ -78,3 +83,34 @@ export function handleMemberTokenRemoved(event: MemberTokenAddedEvent): void {
   let groupMemberId = createSafeGroupMemberId(groupId, safeId)
   store.remove('SafeGroupMember', groupMemberId)
 }
+
+export function handleOnlyOwnerCanMint(event: OnlyOwnerCanMintEvent): void {
+  let groupAddress = event.params._event.address
+  let onlyOwnerCanMint = event.params._onlyOwnerCanMint
+  let groupCurrencyToken = createGroupCurrencyTokenIfNonExistent(groupAddress)
+  groupCurrencyToken.onlyOwnerCanMint = onlyOwnerCanMint
+  groupCurrencyToken.save()
+}
+
+export function handleOnlyTrustedCanMint(event: OnlyTrustedCanMintEvent): void {
+  let groupAddress = event.params._event.address
+  let onlyTrustedCanMint = event.params._onlyTrustedCanMint
+  let groupCurrencyToken = createGroupCurrencyTokenIfNonExistent(groupAddress)
+  groupCurrencyToken.onlyTrustedCanMint = onlyTrustedCanMint
+  groupCurrencyToken.save()
+}
+export function handleOwnerChanged(event: OwnerChangedEvent): void {
+  let groupAddress = event.params._event.address
+  let newOwner = event.params._new
+  let groupCurrencyToken = createGroupCurrencyTokenIfNonExistent(groupAddress)
+  groupCurrencyToken.owner = newOwner.toHexString()
+  groupCurrencyToken.save()
+}
+export function handleSuspended(event: SuspendedEvent): void {
+  let groupAddress = event.params._event.address
+  let groupCurrencyToken = createGroupCurrencyTokenIfNonExistent(groupAddress)
+  // @TODO the suspended event only send the owner address data, it'd be better if send the suspended bool
+  let GroupCurrencyTokenContractInstance = GroupCurrencyTokenContract.bind(groupAddress)
+  groupCurrencyToken.suspended = GroupCurrencyTokenContractInstance.suspended()
```

Comment:
I think the current event `Suspended(owner);`  doesn't make much sense to not include the data of the new status. I think we should push to get this event updated in the contracts to include it, and later update the handler in a new PR when done

Reply:
Totally makes sense, I'll comment this in the group, so we can improve this update in the SG.

Will comment this in the issue: https://github.com/BootNodeDev/circles-groups-safe-app/issues/60